ui/apps/everest/.e2e/release/demand-backup-psmdb-sharding.e2e.ts

```diff
@@ -0,0 +1,344 @@
+        const { items: clusters } = await response.json();
+
+        const addedCluster = clusters.find(
+          (cluster) => cluster.metadata.name === clusterName
+        );
+
+        expect(addedCluster).not.toBeUndefined();
+        expect(addedCluster?.spec.engine.type).toBe(db);
+        expect(addedCluster?.spec.engine.replicas).toBe(size);
+        expect(['600m', '0.6']).toContain(
+          addedCluster?.spec.engine.resources?.cpu.toString()
+        );
+        expect(addedCluster?.spec.engine.resources?.memory.toString()).toBe(
+          '1G'
+        );
+        expect(addedCluster?.spec.engine.storage.size.toString()).toBe('1Gi');
+        expect(addedCluster?.spec.proxy.expose.type).toBe('internal');
+        if (db != 'psmdb') {
+          expect(addedCluster?.spec.proxy.replicas).toBe(size);
+        }
+      });
+    });
+
+    // Add data in MondoDB
+    test(`Add data [${db} size ${size}]`, async () => {
+      await prepareMongoDBTestDB(clusterName, namespace);
+    });
+
+    // Add MongoDB-specific configuration
+    test(`Setup MongoDB-specific sharding [${db} size ${size}]`, async () => {
+      await configureMongoDBSharding(clusterName, namespace);
+    });
+
+    test(`Create demand backup [${db} size ${size}]`, async ({ page }) => {
+      await gotoDbClusterBackups(page, clusterName);
+      await clickOnDemandBackup(page);
+      await page.getByTestId('text-input-name').fill(baseBackupName + '-1');
+      await expect(page.getByTestId('text-input-name')).not.toBeEmpty();
+      await expect(
+        page.getByTestId('text-input-storage-location')
+      ).not.toBeEmpty();
+      await page.getByTestId('form-dialog-create').click();
+
+      await waitForStatus(page, baseBackupName + '-1', 'Succeeded', 500000);
+    });
+
+    test(`Delete data [${db} size ${size}]`, async () => {
+      await dropTestDB(clusterName, namespace);
+    });
+
+    test(`Restore cluster [${db} size ${size}]`, async ({ page }) => {
+      await gotoDbClusterBackups(page, clusterName);
+      await findRowAndClickActions(
+        page,
+        baseBackupName + '-1',
+        'Restore to this DB'
+      );
+      await expect(
+        page.getByTestId('select-input-backup-name')
+      ).not.toBeEmpty();
+      await page.getByTestId('form-dialog-restore').click();
+
+      await page.goto('/databases');
+      await waitForStatus(page, clusterName, 'Restoring', 30000);
+      await waitForStatus(page, clusterName, 'Up', 600000);
+
+      await gotoDbClusterRestores(page, clusterName);
+      // we select based on backup source since restores cannot be named and we don't know
+      // in advance what will be the name
+      await waitForStatus(page, baseBackupName + '-1', 'Succeeded', 120000);
+    });
+
+    test(`Check data after restore [${db} size ${size}]`, async () => {
+      if (db === 'psmdb') {
+        // Validate the data in the database
+        const result = await queryTestDB(clusterName, namespace);
```

Comment:
We can remove this since below we are checking for both t1 and t2 and this is checking for t1 as well.